src/server/apiController.ts

```diff
@@ -1,130 +1,145 @@
-import { Express, Request, Response } from 'express';
-import { injectable } from 'inversify';
-import { Uri } from 'vscode';
+import { Uri, Webview } from 'vscode';
 import { IAvatarProvider } from '../adapter/avatar/types';
 import { GitOriginType } from '../adapter/repository/index';
 import { ICommandManager } from '../application/types/commandManager';
 import { IGitCommitViewDetailsCommandHandler } from '../commandHandlers/types';
 import { CommitDetails, FileCommitDetails } from '../common/types';
 import { IServiceContainer } from '../ioc/types';
-import { Avatar, CommittedFile, IGitService, IGitServiceFactory, LogEntries, LogEntriesResponse, LogEntry, Ref, RefType } from '../types';
-import { IApiRouteHandler } from './types';
+import { Avatar, IGitService, LogEntry, Ref, RefType } from '../types';
 
-// tslint:disable-next-line:no-require-imports no-var-requires
-
-@injectable()
-export class ApiController implements IApiRouteHandler {
+export class ApiController {
     private readonly commitViewer: IGitCommitViewDetailsCommandHandler;
-    constructor(private app: Express,
-        private gitServiceFactory: IGitServiceFactory,
+    constructor(private webview: Webview, private gitService: IGitService,
         private serviceContainer: IServiceContainer,
         private commandManager: ICommandManager) {
 
         this.commitViewer = this.serviceContainer.get<IGitCommitViewDetailsCommandHandler>(IGitCommitViewDetailsCommandHandler);
 
-        this.app.get('/log', this.handleRequest(this.getLogEntries.bind(this)));
+        this.webview.onDidReceiveMessage(this.postMessageParser.bind(this))
+
+        /*this.app.get('/log', this.handleRequest(this.getLogEntries.bind(this)));
         this.app.get('/branches', this.handleRequest(this.getBranches.bind(this)));
         this.app.post('/action/:name?', this.handleRequest(this.doAction.bind(this)));
         this.app.post('/actionref/:name?', this.handleRequest(this.doActionRef.bind(this)));
         this.app.get('/log/:hash', this.handleRequest(this.getCommit.bind(this)));
         this.app.post('/log/:hash', this.handleRequest(this.doSomethingWithCommit.bind(this)));
         this.app.post('/log/:hash/committedFile', this.handleRequest(this.selectCommittedFile.bind(this)));
         this.app.post('/avatars', this.handleRequest(this.getAvatars.bind(this)));
-        this.app.get('/authors', this.handleRequest(this.getAuthors.bind(this)));
+        this.app.get('/authors', this.handleRequest(this.getAuthors.bind(this)));*/
+    }
+
+    private postMessageParser = async (message: any) => {
+        await this[message.cmd](message.args);
```

Comment:
```suggestion
        await this[message.cmd].bind(this)(message.args);
```

In case we use `public async getLogEntries(args: any)` instead of `public getLogEntries = async (args: any)`